src/main/scala/org/bblfsh/client/libuast/org_bblfsh_client_libuast_Libuast.cc

```diff
@@ -143,6 +143,84 @@ JNIEXPORT jobject JNICALL Java_org_bblfsh_client_libuast_Libuast_filter
   return immList;
 }
 
+
+JNIEXPORT jboolean JNICALL Java_org_bblfsh_client_libuast_Libuast_filterBool
+  (JNIEnv *env, jobject self, jobject obj, jstring query) {
+
+  jboolean ret = false;
+  memTracker.EnterFilter();
+
+  try {
+    jobject *node = &obj;
+    const char *cstr = AsNativeStr(query);
+    if (env->ExceptionCheck() == JNI_TRUE || !cstr) {
+      throw std::runtime_error("");
+    }
+
+    bool ok;
+    ret = (jboolean)UastFilterBool(ctx, node, cstr, &ok);
```

Comment:
Could this line be a lambda that is passed to some `filterAny` function, that contain other parts of this code? This will remove duplicated error handling and memory tracking.